napari/layers/intensity_mixin.py

```diff
@@ -43,6 +53,34 @@ def reset_contrast_limits_range(self):
             info = np.iinfo(self.dtype)
             self.contrast_limits_range = (info.min, info.max)
 
+    @property
+    def autoscale_source(self):
+        """Data source for autoscaling, either 'data' (full) or 'slice'."""
+        return self._autoscale_source
+
+    @autoscale_source.setter
+    def autoscale_source(self, source):
+        if source.lower() not in ('data', 'slice'):
+            raise ValueError(
+                "autoscale_source must be either 'data' or 'slice'"
+            )
+        self._autoscale_source = source.lower()
+
+    @property
+    def keep_autoscale(self):
+        """Whether to auto-rescale when the data/view_slice changes."""
+        return self._keep_autoscale
+
+    @keep_autoscale.setter
+    def keep_autoscale(self: 'Image', val):
+        self._keep_autoscale = bool(val)
+        if self._keep_autoscale:
+            self.events.data.connect(self.reset_contrast_limits)
+            self.events.set_data.connect(self.reset_contrast_limits)
+        else:
+            self.events.data.disconnect(self.reset_contrast_limits)
+            self.events.set_data.disconnect(self.reset_contrast_limits)
```

Comment:
I don't really like this internal event connection - we've tried to avoid using events within the layer files and generally i think it's been much better. I guess the alternative is to do the resetting in the slice method instead. it will also be much more clear when it's happening

Reply:
sure, I'll look into an alternative